src/main/java/command/Command.java

```diff
@@ -0,0 +1,18 @@
+package command;
+
+import exception.DukeException;
+import storage.Storage;
+import task.TaskList;
+import ui.Ui;
+
+/**
+ * Abstract class for every Commands.
+ * Encapsulates user's input into a command for easy handling.
+ */
+public abstract class Command {
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    public abstract boolean isExit();

```

Comment:
It might be more efficient to make this class non-abstract and return false. Then in your ExitCommand/ByeCommand, you can override it and make it return true.